packages/react-hooks/src/useMergedRefs.test.tsx

```diff
@@ -1,8 +1,31 @@
 import * as React from 'react';
-import { mount } from 'enzyme';
+import { mount, ReactWrapper } from 'enzyme';
 import { useMergedRefs } from './useMergedRefs';
 
 describe('useMergedRefs', () => {
+  let wrapper: ReactWrapper | undefined;
+
+  afterEach(() => {
+    wrapper?.unmount();
+    wrapper = undefined;
+  });
+
+  it('always returns the same ref (refs should be immutable)', () => {
+    let lastMergedRef;
+
+    const TestComponent: React.FunctionComponent = () => {
+      lastMergedRef = useMergedRefs<boolean>(() => ({}));
+      return null;
+    };
+
+    wrapper = mount(<TestComponent />);
+    const ref1 = lastMergedRef;
+    wrapper.setProps({});
```

Comment:
don't you need to call `wrapper.update` for the component to actually re-render?

Reply:
nope. locally updated to count renders, and it passes.

```
    let lastMergedRef;
    let renderCount = 0;
    const TestComponent: React.FunctionComponent = ({ val }) => {
      renderCount++;
      lastMergedRef = useMergedRefs<boolean>(() => ({}));
      return null;
    };

    wrapper = mount(<TestComponent />);
    const ref1 = lastMergedRef;
    expect(renderCount).toEqual(1);

    wrapper.setProps({});
    expect(renderCount).toEqual(2);
    const ref2 = lastMergedRef;
```

Reply:
In fact, changing that to `update` will update the wrapper state to match the rendered state - not cause a rerender. The test above fails if you change `setProps` to `update`.

Reply:
https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/update.html